### PR TITLE
fix nil pointer issue when validating feature

### PR DESCRIPTION
**Pull request by richardwxn**
since the mtls setting is already removed from default profile, now the validation function would fail to nil pointer issue

### CODE REVIEW

File: operator/pkg/apis/istio/v1alpha1/validation/validation.go [hunk 1]
```diff
@@ -98,17 +98,18 @@ func deprecatedSettingsMessage(values *valuesv1alpha1.Values) string {
...
 	if g == nil {
 		return nil
 	}
-	m := g.GetMtls()
-	if m == nil {
-		return nil
+	mc := iopSpec.MeshConfig
+	val := true
+	if mc != nil && mc["enableAutoMtls"] != nil {
+		val = mc["enableAutoMtls"].(bool)
```
Comment (howardjohn) [inline operator/pkg/apis/istio/v1alpha1/validation/validation.go:110]: can this be a non-bool and panic?

File: operator/pkg/apis/istio/v1alpha1/validation/validation.go [hunk 2]
```diff
@@ -98,17 +98,18 @@ func deprecatedSettingsMessage(values *valuesv1alpha1.Values) string {
...
 	}
-	if m.GetAuto().GetValue() && !g.GetControlPlaneSecurityEnabled().GetValue() {
+	if val && !g.GetControlPlaneSecurityEnabled().GetValue() {
 		return []error{fmt.Errorf("security: auto mtls is enabled, but control plane security is not enabled")}
```
Comment (howardjohn) [inline operator/pkg/apis/istio/v1alpha1/validation/validation.go:113]: can we just remove this whole check? GetControlPlaneSecurityEnabled doesn't actually do anything

File: operator/pkg/apis/istio/v1alpha1/validation/validation.go (hunk 1 above)
Comment (richardwxn) [reply operator/pkg/apis/istio/v1alpha1/validation/validation.go:110]: I assumed we do a type validation first before that, but I can add it for safety

File: operator/pkg/apis/istio/v1alpha1/validation/validation.go (hunk 2 above)
Comment (richardwxn) [reply operator/pkg/apis/istio/v1alpha1/validation/validation.go:113]: you mean the ControlPlaneSecurityEnabled setting? then maybe we should add it to deprecated fields?

**Review by incfly — APPROVED** [review]
I loose track of our install option. if the `values.mtls` is removed, where is the control of `enableAutomTLS` propogated?

**Comment by howardjohn** [comment]
@incfly its controled directly from mesh config with `meshConfig.enableAutoMtls`

File: operator/pkg/apis/istio/v1alpha1/validation/validation.go (hunk 2 above)
Comment (howardjohn) [reply operator/pkg/apis/istio/v1alpha1/validation/validation.go:113]: it controls mixer only actually. So it does do something I guess but not much

Comment (richardwxn) [reply operator/pkg/apis/istio/v1alpha1/validation/validation.go:113]: ah ok then I think we can remove this validation, it creates a lot of warning noise to user

Comment (richardwxn) [reply operator/pkg/apis/istio/v1alpha1/validation/validation.go:113]: but I keep the function there so later we can add other feature validations

**Comment by richardwxn** [comment]
PTAL again: @howardjohn, now the master is broken because of this

**Review by howardjohn — APPROVED** [review]

**Pull request merged**